### ISSUE TITLE
[Feature Request] Create a plugins section on elasticsearch.org

### ISSUE THREAD

**Issue by jerrac** [elasticsearch/elasticsearch#9664]
Apologies if there is a better place for this idea. I didn't see any website repo, and I don't think the docs repo is a good place.

Could we create a section on elasticsearch.org to help us find and evaluate plugins? 

Info on each plugin page:

* Number of people using it. (Not sure how you'd get this data...)
* Issue metadata
** Could you pull in labels like "Security"? So you could see how many issues are feature request, or bugs, or whatever other labels are useful.
* Comments/reviews
* Rating/number of stars
* A "report this plugin" form for use when a plugin is no longer developed but has major security issues. 
* List of similar plugins.

And so on. Basically, a mashup of the kind of information found on galaxy.ansible.com and drupal.org's module pages.

This would:

* Make it easier to find plugins
* Help users evaluate the usefulness/trustworthyness of plugins
* Help resolve issues like #664 where the main answer is to use a plugin, and people like me are resistent to using third-party plugins.

**Comment by markharwood** [comment]
From an end user's perspective I love the idea of there being a curated set of plugins like the one you describe.
However from the perspective of someone who has to maintain the list, I would be nervous about taking on all the issues in ensuring quality of the listings (running tests? security tests? version compatibility checking?) , removing spammers, dealing with plugin authors who are upset that their contributions don't make the cut etc. This is a potentially big undertaking like running the app store which we are not planning to address just now.

We have a basic listing without the quality measures you outlined here: http://www.elasticsearch.org/guide/en/elasticsearch/reference/current/modules-plugins.html

There are independent third parties who try to provide more detailed round-ups e.g. https://blog.codecentric.de/en/2014/03/elasticsearch-monitoring-and-management-plugins/
and I would expect more of these to emerge as time goes on.

**Comment by jerrac** [comment]
>  I would be nervous about taking on all the issues in ensuring quality of the listings (running tests? security tests? version compatibility checking?) , removing spammers, dealing with plugin authors who are upset that their contributions don't make the cut etc.

Would you really need to do all that? Drupal and Ansible appear to have methods that work. 

I would think Ansible's method of connecting the github account would help discourage spam. As in, when their account is blocked on github for spam, it would go away on plugins.es.org as well. 

Actually testing the code for version compatibility and so on, could be left to the community. If something is broken, they report it on the plugins github issue queue. If it doesn't get fixed, then they report it in plugins.es.org's review/feedback feature. No need for you guys to do anything unless you just want to.

So, I think most of the work would just be building and maintaining site infrastructure. The community would take care of evaluating the quality of the listings.

@markharwood Um, out of curiosity, why did you close this issue when there were only two posts? That's not much of a discussion. It seems to me that closing this quickly would discourage people from discussing things like the issues you brought up.

**Comment by s1monw** [comment]
@jerrac we internally discussed this and we really don't have the resources to handle stuff like this at the moment. There is a huge amount of quality work that needs to be done and given the current situation we don't think we can keep up with the demand on what you are suggesting. We closed the issue because there is not much to discuss here as we stated that your idea is great but not realistic at this point since at least between us internally nobody can maintain such a tool. I am sorry if you feel like discouraged but that was not the intention.

**Comment by jerrac** [comment]
@s1monw Ah. It wasn't indicated that there had been an internal discussion. I assume that discussion happened prior to my issue? 

I, personally, wasn't very discouraged. But it took a couple times of something like this happening before I figured out that repo maintainers often close tickets without being clear on why, and that it is fine to keep posting anyway. 

I'd have preferred a closing comment that was something like "We discussed this internally a while back, below is a quick summary of why we can't implement this at this time. Feel free to look into building something like this yourself."

Anyway, thanks for the response.
